File: cpji-jira-plugin-test/src/main/java/com/atlassian/cpji/tests/pageobjects/IssueActionsDialog.scala
package com.atlassian.cpji.tests.pageobjects

import scala.collection.JavaConverters._
import com.atlassian.jira.pageobjects.dialogs.FormDialog
import org.openqa.selenium.By
import com.atlassian.pageobjects.elements.{ElementBy, PageElement}
import com.atlassian.jira.pageobjects.components.fields.{QueryableDropdownSelect, AutoComplete}

class IssueActionsDialog extends FormDialog("issue-actions-dialog") {
	private final val ISSUE_ACTIONS_QUERYABLE_CONTAINER: By = By.id("issueactions-queryable-container")
	private final val ISSUE_ACTIONS: By = By.id("issueactions-suggestions")

	@ElementBy(id = "issueactions-suggestions") protected var suggestions: PageElement = null

	def getAutoComplete: AutoComplete = {
		return binder.bind(classOf[QueryableDropdownSelect], ISSUE_ACTIONS_QUERYABLE_CONTAINER, ISSUE_ACTIONS)
	}

	def getActionsLinksByQuery(query: String): Iterable[String] = {
		getAutoComplete.query(query)
		return suggestions.findAll(By.tagName("a")).asScala.map(_.getAttribute("href"))
	}

	def queryAndSelect[T](search: String, dialogClass: Class[T], args: AnyRef*): T = {
		getAutoComplete.query(search).getActiveSuggestion.click
		return binder.bind(dialogClass, args)
	}
}

File: cpji-jira-plugin-test/src/main/java/com/atlassian/cpji/tests/pageobjects/IssueActionsDialog.java
package com.atlassian.cpji.tests.pageobjects;

import com.atlassian.jira.pageobjects.components.fields.AutoComplete;
import com.atlassian.jira.pageobjects.components.fields.QueryableDropdownSelect;
import com.atlassian.jira.pageobjects.dialogs.FormDialog;
import com.atlassian.pageobjects.elements.ElementBy;
import com.atlassian.pageobjects.elements.PageElement;
import com.google.common.base.Function;
import com.google.common.collect.Iterables;
import org.openqa.selenium.By;

import javax.annotation.Nullable;

/**
 * @since v3.0
 */
public class IssueActionsDialog extends FormDialog {
    public IssueActionsDialog() {
        super("issue-actions-dialog");
    }

    private final By ISSUE_ACTIONS_QUERYABLE_CONTAINER = By.id("issueactions-queryable-container");
    private final By ISSUE_ACTIONS = By.id("issueactions-suggestions");

    @ElementBy(id = "issueactions-suggestions")
    protected PageElement suggestions;

    public AutoComplete getAutoComplete() {
        return binder.bind(QueryableDropdownSelect.class, ISSUE_ACTIONS_QUERYABLE_CONTAINER, ISSUE_ACTIONS);
    }


    public Iterable<String> getActionsLinksByQuery(String query) {
        getAutoComplete().query(query);
        return Iterables.transform(suggestions.findAll(By.tagName("a")), new Function<PageElement, String>() {
            @Override
            public String apply(@Nullable PageElement input) {
                return input.getAttribute("href");
            }
        });
    }

}
